org.mixedrealitytoolkit.spatialmanipulation/ObjectManipulator/ObjectManipulator.cs

```diff
@@ -939,11 +977,38 @@ protected virtual void ModifyTargetPose(ref MixedRealityTransform targetPose, re
 
             bool applySmoothing = ShouldSmooth && smoothingLogic != null;
 
-            targetPose.Position = (applySmoothing && !useForces) ? smoothingLogic.SmoothPosition(HostTransform.position, targetPose.Position, moveLerpTime, Time.deltaTime) : targetPose.Position;
-            targetPose.Rotation = (applySmoothing && !useForces) ? smoothingLogic.SmoothRotation(HostTransform.rotation, targetPose.Rotation, rotateLerpTime, Time.deltaTime) : targetPose.Rotation;
+            targetPose.Position = (applySmoothing && !UseForces) ? smoothingLogic.SmoothPosition(HostTransform.position, targetPose.Position, moveLerpTime, Time.deltaTime) : targetPose.Position;
+            targetPose.Rotation = (applySmoothing && !UseForces) ? smoothingLogic.SmoothRotation(HostTransform.rotation, targetPose.Rotation, rotateLerpTime, Time.deltaTime) : targetPose.Rotation;
             targetPose.Scale = applySmoothing ? smoothingLogic.SmoothScale(HostTransform.localScale, targetPose.Scale, scaleLerpTime, Time.deltaTime) : targetPose.Scale;
         }
 
+        private void UpdateRigidbodyMovementType()
+        {
+            // Find the most recent interactor that has a movement type override, if any, using reverse order of selecting interactors.
+            // This matches XRI's treatment of overrides as the default behavior.
+            MovementType? movementTypeOverride = null;
```

Comment:
We can avoid the overhead of boxing a value type...  I believe you can refactor to this...

```cs
            currentRigidbodyMovementType = rigidbodyMovementType;
            for (var index = interactorsSelecting.Count - 1; index >= 0; --index)
            {
                var xrBaseInteractor = interactorsSelecting[index] as XRBaseInteractor;
                if (xrBaseInteractor != null && xrBaseInteractor.selectedInteractableMovementTypeOverride.HasValue)
                {
                    currentRigidbodyMovementType = xrBaseInteractor.selectedInteractableMovementTypeOverride.Value;
                    break;
                }
            }
```

Reply:
Good point!  Done.